website/blog/2024-03-20-snapshot-testing.md

```diff
@@ -0,0 +1,173 @@
+```
+
+This makes maintaining your tests so much easier. The same works for all other types of objects, e.g. CSS Properties, or the text content of an element. They all can be converted into a snapshot to simplify the assertion and keep your tests lean. This can also speed up your tests by merging many single assertions into one:
+
+```ts
+const elem = $('#alertBar')
+await expect(elem).toHaveAttribute('data-alert')
+await expect(elem).toHaveCSSProperty('background-color', 'green')
+await expect(elem).toHaveText('You logged into a secure area!')
+```
+
+now becomes a single:
+
+```ts
+await expect($('#alertBar')).toMatchSnapshot()
+/**
+ * stores the following into a snapshot file:
+ *
+ * <div data-alert="" id="flash" class="flash success">
+ *   You logged into a secure area!
+ *   <a href="#" class="close">×</a>
+ * </div>
+ */
+```
+
+You can find more information about DOM and object-based snapshots in our [Snapshot guide](/docs/snapshot).
+
+## Visual Snapshots
+
+While taking snapshots of an element structure and its attributes might be great and powerful, it comes with an important caveat: even though we are testing that the element has a class name called `success`, this doesn't guarantee that the alert is green!
+
+For these reasons, visual testing has become a very popular tool as it includes how elements are rendered, in which color and can ensure that e.g. it is not overlaid by any other element. Taking visual snapshots works very similar, as you can:
+
+- take a visual snapshot of the whole screen:
+  ```ts
+  await expect(browser).toMatchScreenSnapshot('partialPage')
+  ```
+- take a visual snapshot of an element:
+  ```ts
+  await expect($('#element-id')).toMatchElementSnapshot('firstButtonElement')
+  ```
+- take a snapshot of the whole page:
+  ```ts
+  await expect(browser).toMatchFullPageSnapshot('fullPage')
+  ```
+- or take a page snapshot that includes [page tab-ability](https://vivrichards.co.uk/accessibility/automating-page-tab-flows-using-visual-testing-and-javascript):
+  ```ts
+  await expect(browser).toMatchTabbablePageSnapshot('check-tabbable')
+  ```
+
+WebdriverIO will store these visual snapshots conveniently next to your text-based snapshots within the `__snapshots__` directory next to your test.
+
+While text-based snapshot testing is built into WebdriverIO, you have to install a [service](https://www.npmjs.com/package/@wdio/visual-service) to enable all visual snapshot capabilities via:
+
+```sh
+npm i --save-dev @wdio/visual-service
+```
+
+With the most recent release of the Visual Testing Module with have shipped further improvements for **Native App Snapshot Testing**.
+
+### Native App Snapshot Testing
+
+The module now supports `saveElement`, `checkElement`, `saveScreen`, and `checkScreen` methods, along with `toMatchElementSnapshot` and `toMatchScreenSnapshot` matchers for native apps. It automatically detects the testing context (web, webview, or native_app) to streamline your workflow.
```

Comment:
The blog never talked about the "internal" methods, so maybe adjust it to the following

```suggestion
The module now supports the `toMatchElementSnapshot` and `toMatchScreenSnapshot` matchers for Mobile native apps. It automatically detects the testing context (web, webview, or native_app) to streamline your workflow.
```